pkg/streamingpromql/types/fpoint_ring_buffer.go

```diff
@@ -294,6 +304,137 @@ func (v FPointRingBufferView) Clone() (*FPointRingBufferView, *FPointRingBuffer,
 	return view, buffer, nil
 }
 
+func (v FPointRingBufferView) Iterator(existing *FPointRingBufferViewIterator) *FPointRingBufferViewIterator {
+	if existing == nil {
+		existing = &FPointRingBufferViewIterator{}
+	}
+	existing.idx = 0
+	existing.view = &v
```

Comment:
Rather than taking the address like this, I would change this (and all other `FPointRingBufferView` methods) to take a pointer receiver, then we can assign `v` as-is here - we always return a pointer from the `View...` methods, so this shouldn't change much.